tests/php/json-api/test-class.json-api-update-post-endpoints.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * Jetpack `sites/%s/posts/%d` endpoint unit tests.
+ *
+ * @package Jetpack
+ */
+
+require_jetpack_file( 'class.json-api-endpoints.php' );
+
+/**
+ * Jetpack `sites/%s/posts/%d` endpoint unit tests.
+ */
+class WP_Test_Json_Api_Update_Post_Endpoints extends WP_UnitTestCase {
+
+	/**
+	 * Inserts globals needed to initialize the endpoint.
+	 */
+	private function set_globals() {
+		$_SERVER['REQUEST_METHOD'] = 'Post';
+		$_SERVER['HTTP_HOST']      = '127.0.0.1';
+		$_SERVER['REQUEST_URI']    = '/';
+	}
+
+	/**
+	 * Prepare the environment for the test.
+	 */
+	public function setUp() {
+		global $blog_id;
+
+		if ( ! defined( 'WPCOM_JSON_API__BASE' ) ) {
+			define( 'WPCOM_JSON_API__BASE', 'public-api.wordpress.com/rest/v1' );
+		}
+
+		parent::setUp();
+
+		$this->set_globals();
+
+		WPCOM_JSON_API::init()->token_details = array( 'blog_id' => $blog_id );
+	}
+
+	/**
+	 * Unit tests the should untrash post protected method.
+	 */
+	public function test_update_post_api_v1_1() {
+
+		$post_id  = $this->get_post();
+		$endpoint = $this->get_endpoint();
+
+		$post = get_post( $post_id );
+
+		$this->assertTrue( $this->invoke_method( $endpoint, 'should_untrash_post', array( 'trash', 'draft', $post ) ) );
+		$this->assertFalse( $this->invoke_method( $endpoint, 'should_untrash_post', array( 'publish', 'trash', $post ) ) );
+		$this->assertFalse( $this->invoke_method( $endpoint, 'should_untrash_post', array( 'publish', 'draft', $post ) ) );
+	}
+
+	/**
+	 * Unit tests the untrash post protected method.
+	 */
+	public function test_update_post_api_v1_1_untrash_post() {
+		$post_id  = $this->get_post();
+		$endpoint = $this->get_endpoint();
+
+		wp_trash_post( $post_id );
+		$post  = get_post( $post_id );
+		$input = array( 'slug' => 'hello__trashed' );
```

Comment:
It is not obvious from where `hello` string coming from

Reply:
The hello is coming from the generated post that  gets create in the get_post() method.

Reply:
Yeah, I get that. It just not obvious right a way